src/imports.rs

```diff
@@ -39,6 +35,24 @@ impl<'a> FmtVisitor<'a> {
             _ => ""
         };
 
+        if path_list.len() == 1 {
+            let vpi = path_list[0];
+            let rest = match vpi.node {
+                ast::PathListItem_::PathListIdent{ name, .. } => {
+                    token::get_ident(name).to_string()
+                }
+                ast::PathListItem_::PathListMod{ .. } => "self".to_string()
```

Comment:
I'm not sure quite what to do here. It catches things like ``use blah::{self}`` which should become ``use::blah`` I think. I was going to write:

```
ast::PathListItem_::PathListMod{ .. } => {
    if path_str.len() != 0 {
        format!("{}use {};", vis, path_str);
    }
     panic!("Error: use {self};");
}
```

But I wasn't sure what you wanted to do about error handling.



I'm really sorry for all the faff!